client/src/controllers/SwapController.ts

```diff
@@ -197,24 +203,29 @@ export class SwapController extends Controller<
     });
   }
 
+  get formElement() {
+    return (
+      this.hasInputTarget ? this.inputTarget.form || this.element : this.element
+    ) as HTMLFormElement;
+  }
+
   /**
    * Update the target element's content with the response from a request based on the input's form
    * values serialised. Do not account for anything in the main location/URL, simply replace the content within
    * the target element.
    */
   submit() {
-    const form = (
-      this.hasInputTarget ? this.inputTarget.form : this.element
-    ) as HTMLFormElement;
-
-    // serialise the form to a query string
-    // https://github.com/microsoft/TypeScript/issues/43797
-    const searchParams = new URLSearchParams(new FormData(form) as any);
+    const form = this.formElement;
+    const data = new FormData(form);
 
+    // serialise the form to a query string if it's a GET request
+    // cast as any to avoid https://github.com/microsoft/TypeScript/issues/43797
+    const searchParams = new URLSearchParams(data as any);
```

Comment:
I think it would be clearer if this was done conditionally? As-is the comment is incorrect, the serialisation always happens, it just ends up being unused for other methods